### PR TITLE
Add initial SVS-based reimplementation and Nix development shell

### PR DESCRIPTION
**Pull request by tanim-ics**
This PR intends to adopt the State Vector Sync Protocol instead of the Partial PSync protocol.

### CODE REVIEW

File: include/iceflow/IceFlowPubSubBase.hpp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    secOpts.interestSigner->signingInfo.setSigningHmacKey(
+        "dGhpcyBpcyBhIHNlY3JldCBtZXNzYWdl");
+
+    // Sign data packets using SHA256 (for simplicity)
+    secOpts.dataSigner->signingInfo.setSha256Signing();
+
+    // Do not fetch publications older than 10 seconds
+    ndn::svs::SVSPubSubOptions opts;
+    opts.UseTimestamp = true;
+    opts.MaxPubAge = ndn::time::milliseconds(10000);
```
Comment (JKRhb) [inline include/iceflow/IceFlowPubSubBase.hpp:47]: The CI is complaining about these:

```suggestion
    opts.useTimestamp = true;
    opts.maxPubAge = ndn::time::milliseconds(10000);
```

File: ansible-scripts/dependencySolver.yaml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    - name: Check if NFD-out-of-the-box directory exists
+      stat:
+        path: "{{ dest }}/NFD-out-of-the-box"
+      register: NFD_directory
+
+    - name: Clone NFD-out-of-the-box from GitHub
+      git:
+        repo: https://github.com/JKRhb/NFD-out-of-the-box.git
+        dest: "{{ dest }}/NFD-out-of-the-box"
+        version: master
```
Comment (JKRhb) [inline ansible-scripts/dependencySolver.yaml:124]: I think this has to be the `self-learning-rebased` branch, right?

Comment (tanim-ics) [reply ansible-scripts/dependencySolver.yaml:124]: ah yes. thank you 👍🏽

**Comment by JKRhb** [comment]
Since we're switching the CI pipeline to a Nix-based approach in #47, I am going to merge this one despite the failed checks.

**Pull request merged**